### DISCUSSION TITLE
Feature: Timestamp Template

### DISCUSSION THREAD

**Discussion by iris2jane** [aidenlx/media-extended#650] [Ideas]
I would like to automatically add some fields, such as Title Level 6("###### "), before the timestamp, which may make the video note level clearer.

***

If there are more feature request around templates, you can reply directly in this issue.

**Comment by Tony-zf1** [comment]
@aidenlx 作者你好，我想要的自定义时间戳格式的功能比较简单。
当前的时间戳格式是：`回车`+`[03:05](http://***)`+`回车`。
在写大纲笔记（workflowy或幕布的笔记形式）时，我希望时间戳直接在行尾，以保证笔记的大纲格式不被破坏。
因此希望这个时间戳格式可以自定义为：`[03:05](http://***)`+`回车`。
视频说明在[b站](https://www.bilibili.com/video/BV1z64y1x7ki)，谢谢!

**Comment by iris2jane** [comment]
> @aidenlx 作者你好，我想要的自定义时间戳格式的功能比较简单。
> 当前的时间戳格式是：`回车`+`[03:05](http://***)`+`回车`。
> 在写大纲笔记（workflowy或幕布的笔记形式）时，我希望时间戳直接在行尾，以保证笔记的大纲格式不被破坏。
> 因此希望这个时间戳格式可以自定义为：`[03:05](http://***)`+`回车`。
> 视频说明在[b站](https://www.bilibili.com/video/BV1z64y1x7ki)，谢谢!

插件设置里现在可以自定义了

**Comment by aidenlx** [comment]
> @aidenlx 作者你好，我想要的自定义时间戳格式的功能比较简单。
> 当前的时间戳格式是：`回车`+`[03:05](http://***)`+`回车`。
> 在写大纲笔记（workflowy或幕布的笔记形式）时，我希望时间戳直接在行尾，以保证笔记的大纲格式不被破坏。
> 因此希望这个时间戳格式可以自定义为：`[03:05](http://***)`+`回车`。
> 视频说明在[b站](https://www.bilibili.com/video/BV1z64y1x7ki)，谢谢!

在v2.9.0，你可以在设置里只留下`{{TIMESTAMP}}`，不保留换行符，然后手动回车，因为obsidian要触发自动缩进需要手动按回车键，插件直接操作编辑器是没法触发的

![image](https://user-images.githubusercontent.com/31102694/127014263-4a295cc2-ae2c-4f5f-9103-9bcddb0367cd.png)

PS: 最好以后GitHub上附一份英文（百度翻译也可以），要不成国人的加密通讯了😂

**Comment by Tony-zf1** [comment]
> 
> 在v2.9.0，你可以在设置里只留下`{{TIMESTAMP}}`，不保留换行符，然后手动回车，因为obsidian要触发自动缩进需要手动按回车键，插件直接操作编辑器是没法触发的
> 
> ![image](https://user-images.githubusercontent.com/31102694/127014263-4a295cc2-ae2c-4f5f-9103-9bcddb0367cd.png)
> 
> PS: 最好以后GitHub上附一份英文（百度翻译也可以），要不成国人的加密通讯了😂

Thank you so much! The template works well as below.

![image](https://user-images.githubusercontent.com/79450935/127017568-32142d51-3d02-47cf-ad7f-67f402980cd0.png)

A cup of MIXUE Icecream&Tea has been offered with Alipay. 

Thanks again!

PS：一开始是发在b站插件那边，想着没老外看就直接打的中文，到这边补充也没换成英文😂

**Comment by aidenlx** [comment]
Moved from #81 created by  @dummifiedme

# More templates especially for Cursor control

When I am taking notes, its essential that we can have cursor control to move the cursor to a location after inserting time link.

for example:

I mostly want my notes to be of the form
`- Some text {{timestamp}}.`

While doing this, if I insert timestamp, I need to then go back to the staarting point and then type the thing. If it was possible to have a template for cursor then we could have a template like : `{{cursor}}{{TIMESTAMP}}`

It would help other users too depeding on how they take and format note :)

Templater plugin has this function called `tp.file.cursor()` for the same thing, maybe something like take could be explored.

**Comment by geekhuashan** [comment]
> 

yes,yes. 

I am a novice using obsidian,
And how can I ues Tamplater to get this effect.  
I just creat a template with the <%  tp.file.cursor(){{TIMESTAMP}}  %>

Then failed.

Which command should I generate to  achieve this effect.


Looking forward to your kindly reply

**Comment by chaecramb** [comment]
I would love to have a {{TIME}} template that just gives the time alone. This is so that I can setup a template for local files that just displays the time, in the same way as {{TIMESTAMP}} works for YouTube. 

Currently with local files we get:

`[[filname.mp3#t=00:10.969]] `

I’d like to be able to configure to be consistent with YouTube timestamps, so that we get:

`[10:96](filname.mp3#t=00:10.969)`

Perhaps by separating out {{TIME}} and {{LOCATION}} would create more flexibility.

**Comment by mdmanurung** [comment]
> Moved from #81 created by @dummifiedme
> # More templates especially for Cursor control
> 
> When I am taking notes, its essential that we can have cursor control to move the cursor to a location after inserting time link.
> 
> for example:
> 
> I mostly want my notes to be of the form `- Some text {{timestamp}}.`
> 
> While doing this, if I insert timestamp, I need to then go back to the staarting point and then type the thing. If it was possible to have a template for cursor then we could have a template like : `{{cursor}}{{TIMESTAMP}}`
> 
> It would help other users too depeding on how they take and format note :)
> 
> Templater plugin has this function called `tp.file.cursor()` for the same thing, maybe something like take could be explored.

giving another upvote for this.

**Comment by JacobLightman** [comment]
#### More templates for cursor control

I'd like to automatically have my cursor _after_ the timestamp without having to use another (templater) hotkey. Currently in Media Extended I have {{TIMESTAMP}} <%tp.file.cursor() %> but this requires the use of two hotkeys (Shift + Space, Ctrl + Space) to write a note with a timestamp while I'm watching a video, and sometimes I take _a lot_ of timestamped notes without pausing, which makes this frustrating. If there was a way to achieve this with one hotkey it would be fantastic.
